chain/neutrino.go

```diff
@@ -785,3 +782,17 @@ out:
 	close(s.dequeueNotification)
 	s.wg.Done()
 }
+
+// getNewRescanner injects the Rescanner constructor when called and defaults
+// to using neutrino.NewRescan when unspecified.
+func (s *NeutrinoClient) getNewRescanner() rescan.New {
```

Comment:
I think this function should be passed in when creating the client. See my main comment with the proposed diff.